phoenix-core-client/src/main/java/org/apache/phoenix/util/UpgradeUtil.java

```diff
@@ -1530,15 +1531,23 @@ public static void moveHBaseLevelTTLToSYSCAT(PhoenixConnection oldMetaConnection
                             byte[] tableName = rowKeyMetaData[PhoenixDatabaseMetaData.TABLE_NAME_INDEX];
 
                             String fullTableName = SchemaUtil.getTableName(schemaName, tableName);
+                            currentTableName = fullTableName;
                             if (SchemaUtil.isSystemTable(SchemaUtil.getTableNameAsBytes(schemaName, tableName))) {
                                 //We do not support system table ttl through phoenix ttl, and it will be moved to a
                                 //constant value in future commit.
                                 continue;
                             }
+                            PTable pTable = oldMetaConnection.getTable(null, fullTableName);
```

Comment:
I wonder if this could cause too many getTable() calls to syscat, though should be manageable I assume. WDYT @jpisaac @tkhurana?

Reply:
@virajjasani  Do not think it will be a lot ~ 200 calls during upgrade. The SYSCAT has been used to large number of getTable calls during client startup anyways.